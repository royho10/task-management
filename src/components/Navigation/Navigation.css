* {
	box-sizing: border-box;
}

body {
	font-family: 'Montserrat', sans-serif;
	line-height: 1.6;
	margin: 0;
	min-height: 100vh;
}

ul {
	margin: 0;
	padding: 0;
	list-style: none;
}

.nav-bar {
	z-index: 10;
	position: fixed;
	top: 0;
	width: 100%;
	height: 60px;
	display: flex;
	justify-content: space-between;
	align-items: center;
	padding-top: .5em;
	padding-bottom: .5em;
	border: 1px solid #a2a2a2;
	background-color: #f4f4f4;
	-webkit-box-shadow: 0px 0px 14px 0px rgba(0,0,0,0.75);
	-moz-box-shadow: 0px 0px 14px 0px rgba(0,0,0,0.75);
	box-shadow: 0px 0px 14px 0px rgba(0,0,0,0.75);
	-webkit-border-radius: 5px;
	-moz-border-radius: 5px;
	border-radius: 5px;	
}

.logo {
	margin: 0;
	font-size: 1.45em;
	color: #34495e;	
} 

.logo,
.main-nav {	
	padding: 10px 15px;
	align-items: center;
	text-align: center;
	display: flex;
}

.main-nav {
	color: #34495e;
	font-size: 1em;
	margin-top: 5px;
	text-transform: uppercase;	
}

.main-nav:hover {
	color: #718daa;
}

