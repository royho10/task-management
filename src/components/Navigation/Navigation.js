import React from 'react';
import './Navigation.css';

const Navigation = ({ onRouteChange, isSignedIn }) => {
	if (isSignedIn) {
		return (
			<div className="header">
				<nav className="nav-bar">
					<h1 className="logo">TaskIT</h1>
					<ul className="main-nav">
						<li className="link dim pointer" onClick={() => onRouteChange('signin')}>Log Out</li>
					</ul>
				</nav>
			</div>
		)
	} else {
		return (
			<nav className="nav-bar">
				<h1 className="logo">TaskIT</h1>
				<ul className="main-nav">
					<li className="link dim pointer mr4" onClick={() => onRouteChange('signin')}>Sign In</li>
					<li className="link dim pointer" onClick={() => onRouteChange('resgister')}>Register</li>
				</ul>
			</nav>			
		)
	}
}

export default Navigation;