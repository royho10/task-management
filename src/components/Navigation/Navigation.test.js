import { shallow } from 'enzyme';
import React from 'react';
import Navigation from './Navigation';

it('expect to render Navigation component', () => {
	
	expect(shallow(<Navigation />)).toMatchSnapshot();
})