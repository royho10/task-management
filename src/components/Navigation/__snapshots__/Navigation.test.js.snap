// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render Navigation component 1`] = `
<nav
  className="nav-bar"
>
  <h1
    className="logo"
  >
    TaskIT
  </h1>
  <ul
    className="main-nav"
  >
    <li
      className="link dim pointer mr4"
      onClick={[Function]}
    >
      Sign In
    </li>
    <li
      className="link dim pointer"
      onClick={[Function]}
    >
      Register
    </li>
  </ul>
</nav>
`;
